syntax = "proto3";

package lark;

message PianoRoll {
    string id = 1;
    string collection_name = 2;
    string filename = 3;
    int32 resolution = 4;
    int64 total_tick = 5;
    repeated Track tracks = 6;

    message Track {
        string name = 1;
        repeated Note notes = 2;

        message Note {
            int32 velocity = 1;
            int32 pitch = 2;
            int64 start_tick = 3;
            int64 end_tick = 4;
        }
    }
}

//protoc -I=/Users/Zongyu/Desktop/MYFP/protobuf --python_out=/Users/Zongyu/Desktop/MYFP/protobuf /Users/Zongyu/Desktop/MYFP/protobuf/music.proto
